### PR TITLE
Tom/update state docs

### PR DESCRIPTION
**Pull request by tgberkeley**
Added new var operations page and gave examples of all of the different var operations in use.

### CODE REVIEW

**Comment by tgberkeley** [comment]
One more thing I think we could add is more examples of comparing var operations on the front-end to doing the same thing with a computed var on the backend and when we want to do either approach

File: docs/state/var-operations.md [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+
+class VarComparisonState(State):
+    number_1: int
+    number_2: int
+
+    def update(self):
+        self.number_1 = random.randint(-10, 10)
+        self.number_2 = random.randint(-10, 10)
+
+def var_comparison_example():
```
Comment (Alek99) [inline docs/state/var-operations.md:176]: Lets make this display into a table for a clean visual look

**Review by Alek99 — CHANGES_REQUESTED** [review]

File: docs/state/var-operations.md [hunk 2]
```diff
@@ -0,0 +1,529 @@
...
+        self.number_1 = round(random.uniform(5.1, 9.9), 2)
+        self.number_2 = round(random.uniform(0.1, 4.9), 2)
+
+def var_div_example():
+    return rx.vstack(
+        rx.table_container(
+            rx.table(
+                headers=["Integer 1", "Integer 2", "Operation", "Outcome"],
+                rows=[
+                    (VarDivState.number_1, VarDivState.number_2, "True Division of Int 1 by Int 2 (/)", f"{VarDivState.number_1 / VarDivState.number_2}"),
```
Comment (Alek99) [inline docs/state/var-operations.md:227]: Same here about not needing the comment, as these are regular Python operations

File: docs/state/var-operations.md [hunk 3]
```diff
@@ -0,0 +1,529 @@
...
+    list_2: list = [7, 8, 9, 10]
+    list_3: list = ["p","y","t","h","o","n"]
+
+def var_list_example():
+    return rx.vstack(
+        rx.heading(f"List 1: {VarListState.list_1}", size="lg"),
+        rx.heading(f"List 2: {VarListState.list_2}", size="lg"),
+        rx.heading(f"List 3: {VarListState.list_3}", size="lg"),
+        
+        rx.heading(f"Does List 1 contain the number 3: {VarListState.list_1.contains(3)}", size="md"),
```
Comment (Alek99) [inline docs/state/var-operations.md:305]: I think a clean way to display this section would be to have an 

hstack(
vstack(
rx.heading(f"List 1: {VarListState.list_1}", size="lg"),
rx.heading(f"Contains 3: {VarListState.list_1.contains(3)}", size="md"),
),
vstack(
rx.heading(f"List 2: {VarListState.list_2}", size="lg"),
rx.heading(f"Reverse List 2: ...),
)
...
)
 
 To better show the correlated parts. 
 
 Can shorten these comments
 Does List 1 contain the number 3: true  -> List 1 Contains 3: True
Join List 3 into string: python -> List 3 Joins: python

File: docs/state/var-operations.md [hunk 4]
```diff
@@ -0,0 +1,529 @@
...
+    string_1: str = "PYTHON is FUN"
+    string_2: str = "react is hard"
+   
+
+def var_string_example():
+    return rx.vstack(
+        rx.heading(f"List 1: {VarStringState.string_1}", size="lg"),
+        rx.heading(f"List 2: {VarStringState.string_2}", size="lg"),
+
+        rx.heading(f"Lower Case output of List 1: {VarStringState.string_1.lower()}", size="md"),
```
Comment (Alek99) [inline docs/state/var-operations.md:333]: Same comment about how to strucutre this section with hstack and vstacks as Contains, Reverse and Join section,

Can shorten the comments up to 

List 1 Lower Case: python is fun
List 2 Upper Case: REACT IS HARD

**Review by Alek99 — CHANGES_REQUESTED** [review]
Nice left a few comments about how we can make the examples more concise

File: docs/state/var-operations.md [hunk 5]
```diff
@@ -0,0 +1,529 @@
...
+    numbers_seen: list = []
+    def update(self):
+        self.number = random.randint(-100, 100)
+        self.numbers_seen.append(self.number)
+
+def var_operation_example():
+    return rx.vstack(
+        rx.heading(f"The number: {VarOperationState.number}", size="lg"),
+    
+        rx.heading(f"The negated number is {-VarOperationState.number}", size="md"),
```
[the remark this answers is not in the document]
Comment (tgberkeley) [reply docs/state/var-operations.md:148]: Done, but I feel the numbers look a little small now

File: docs/state/var-operations.md [hunk 6]
```diff
@@ -0,0 +1,529 @@
...
+
+def var_comparison_example():
+    
+    return rx.vstack(
+        rx.table_container(
+            rx.table(
+                headers=["Integer 1", "Integer 2", "Operation", "Outcome"],
+                rows=[
+                    (VarComparisonState.number_1, VarComparisonState.number_2, "are equal (==)", f"{VarComparisonState.number_1 == VarComparisonState.number_2}"),
+                    (VarComparisonState.number_1, VarComparisonState.number_2, "are not equal (!=)", f"{VarComparisonState.number_1 != VarComparisonState.number_2}"),
```
[the remark this answers is not in the document]
Comment (tgberkeley) [reply docs/state/var-operations.md:184]: done

File: docs/state/var-operations.md [hunk 7]
```diff
@@ -0,0 +1,529 @@
...
+### Comparisons and Mathematical Operators
+
+All of the comparison operators are used as expected in python. These include `==`, `!=`, `>`, `>=`, `<`, `<=`. 
+
+There are operators to add two vars `+`, subract two vars `-`, multiply two vars `*` and raise a var to a power `pow()`.
+
+```python exec
+import random
+
+class VarComparisonState(State):
```
[the remark this answers is not in the document]
Comment (tgberkeley) [reply docs/state/var-operations.md:168]: Done

File: docs/state/var-operations.md (hunk 3 above)
Comment (tgberkeley) [reply docs/state/var-operations.md:305]: done

File: docs/state/var-operations.md (hunk 4 above)
Comment (tgberkeley) [reply docs/state/var-operations.md:333]: done

**Comment by Alek99** [comment]
Nice much clearer now 👍

**Review by Alek99 — APPROVED** [review]

**Pull request merged**